src/libvirtApi/domain.js

```diff
@@ -1089,3 +1089,29 @@ export async function domainAddTPM({ connectionName, vmName }) {
     const args = ["virt-xml", "-c", `qemu:///${connectionName}`, "--add-device", "--tpm", "default", vmName];
     return cockpit.spawn(args, { err: "message", superuser: connectionName === "system" ? "try" : null });
 }
+
+export function domainAttachVnc({ connectionName, vmName, vncAddress, vncPort, vncPassword }) {
+    const args = ['virt-xml', '-c', `qemu:///${connectionName}`, vmName, '--add-device', '--graphics', `vnc,listen=${vncAddress},port=${vncPort},passwd=${vncPassword}`];
```

Comment:
I'm not sure if the tests cover this, as you said that cockpit never asks for the password anyway: For an empty password, is `passwd=` equivalent to not providing the `passwd` option at all? (It's not for /etc/shadow)

Reply:
Empty password in XML (`passwd=''` attribute) prevents any VNC connection. (And cockpit crashes, have to figure that out.)

Empty password with virt-xml means "connection allowed without password". virt-xml will omit the passwd attribute in the XML completely. So we do it correctly here.

Reply:
Great, thanks for checking!